index.js

```diff
@@ -0,0 +1,306 @@
+      operand: value.getUint16(2, true),
+    };
+    console.log('RACP Event:', this.racpObject);
+
+    switch (this.racpObject.opCode) {
+      case 0x05:
+        self.emit('numberOfRecords', this.racpObject.operand);
+        break;
+      case 0x06:
+        if (this.racpObject.operand === 0x0101) {
+          console.log('Success.');
+          self.emit('data', self.records);
+        }
+        break;
+      default:
+        throw Error('Unrecognized op code');
+    }
+  }
+
+  static parseGlucoseMeasurement(result) {
+    const record = {
+      flags: result.getUint8(0),
+      seqNum: result.getUint16(1, true),
+    };
+    let offset = 0;
+
+    const dateTime = {
+      year: result.getUint16(3, true),
+      month: result.getUint8(5),
+      day: result.getUint8(6),
+      hours: result.getUint8(7),
+      minutes: result.getUint8(8),
+      seconds: result.getUint8(9),
+    };
+
+    if (this.hasFlag(FLAGS.TIME_OFFSET_PRESENT, record.flags)) {
+      record.payload = {
+        internalTime: sundial.buildTimestamp(dateTime),
+        timeOffset: result.getInt16(10, true),
+      };
+      record.timestamp = sundial.applyOffset(
+        record.payload.internalTime,
+        record.payload.timeOffset,
+      );
+      offset += 2;
+    } else {
+      record.timestamp = sundial.buildTimestamp(dateTime);
+    }
+
+    if (this.hasFlag(FLAGS.GLUCOSE_PRESENT, record.flags)) {
+      if (this.hasFlag(FLAGS.IS_MMOL, record.flags)) {
+        record.units = 'mmol/L';
+      } else {
+        record.units = 'mg/dL';
+      }
+      record.value = this.getSFLOAT(result.getUint16(offset + 10, true), record.units);
+      record.type = result.getUint8(offset + 12) >> 4;
+      record.location = result.getUint8(offset + 12) && 0x0F;
+
+      if (this.hasFlag(FLAGS.STATUS_PRESENT, record.flags)) {
+        record.status = result.getUint16(offset + 13, true);
+      }
+    } else {
+      console.log('No glucose value present for ', sundial.formatDeviceTime(record.timestamp));
+    }
+
+    return record;
+  }
+
+  static getSFLOAT(value, units) {
+    const SFLOAT = {
+      POSITIVE_INFINITY: 0x07FE,
+      NAN: 0x07FF,
+      NRES: 0x0800,
+      RESERVED_VALUE: 0x0801,
+      NEGATIVE_INFINITY: 0x0802,
+    };
+
+    const RESERVED_FLOAT_VALUES = [
+      Number.POSITIVE_INFINITY,
+      NaN,
+      NaN,
+      NaN,
+      Number.NEGATIVE_INFINITY,
+    ];
+
+    let exponent = value >> 12;
+    let mantissa = value & 0x0FFF;
+
+    if (exponent >= 0x0008) {
+      exponent = -((0x000F + 1) - exponent);
+    }
+
+    if (units === 'mg/dL') {
+      exponent += 5; // convert kg/L to mg/dL
+    } else if (units === 'mmol/L') {
+      exponent += 3; // convert mol/L to mmol/L
+    } else {
+      throw Error('Illegal units for glucose value');
+    }
+
+    if (mantissa >= SFLOAT.POSITIVE_INFINITY &&
+        mantissa <= SFLOAT.NEGATIVE_INFINITY) {
+      return RESERVED_FLOAT_VALUES[mantissa - SFLOAT.POSITIVE_INFINITY];
+    }
```

Comment:
When comparing to the reserved SFLOAT values, the exponent needs to be 0. It may be better to do this comparison before separating the exponent and mantissa (unless I have a misunderstanding of your code).

Reply:
If you're finding my code confusing, that probably means it should be rewritten :wink: I'm going to create a new PR with the suggested changes soon.